crates/storage/provider/src/traits/chain.rs

```diff
@@ -139,3 +139,42 @@ impl CanonStateNotification {
         receipts
     }
 }
+
+/// A type that allows to register finalized block related event subscriptions
+/// and get notified when a new finalized block header is available.
+pub trait FinalizedBlockHeaderSubscription: Send + Sync {
+    /// Get notified when a new finalized block header is available.
+    fn subscribe_to_finalized_header(&self) -> broadcast::Receiver<SealedHeader>;
```

Comment:
I'd like to roll a standalone wrapper type for this, because perhaps we want to put some more features into this,

I know we didn't do this for `CanonStateNotifications` but we should have